### DISCUSSION TITLE
Show version available of library added as a constraint

### DISCUSSION THREAD

**Discussion by juancarlosjr97** [renovatebot/renovate#27042] [Suggest an Idea]
Consider a scenario where we are examining projects relying on tools like Prettier. Suppose a project has a peer dependency on Prettier@2, preventing an upgrade to Prettier v3. Now, when the library with this peer dependency is eventually updated to use Prettier@3, Renovate faces a challenge. Due to the existing constraint (dependency rule) in Renovate, it cannot automatically upgrade to Prettier v3.

While Renovate may not be able to perform the update, my suggestion is to still recognize and display this information in the Dependency Dashboard. It could present a table showcasing available versions for the dependency rule or constraint library. This way, users can easily identify the available versions, ensuring that information about a new version is not concealed but remains accessible.

**Comment by secustor** [comment]
What kind of rule do you have in place right now?    
I guess you simply use `"enabled": false`?

Use `dependencyDashboardApproval` or simply closing the PR and there for ignoring the major version. Both will be visible in the DependencyDashboard

**Reply by juancarlosjr97** [reply]
If you look at this example with these rules https://github.com/juancarlosjr97/renovate-test-constraints/blob/main/renovate.json. It identifies node-fetch but would be nice to say somewhere that v3 is also available https://github.com/juancarlosjr97/renovate-test-constraints/pull/2.

**Reply by viceice** [reply]
use dependency dashboard approval instead of allowed version. then you will see the update on dashboard

**Reply by HonkingGoose** [reply]
@juancarlosjr97 I recommend you read the [Renovate docs about the Dependency Dashboard Approval workflow](https://docs.renovatebot.com/key-concepts/dashboard/#dependency-dashboard-approval-workflow) to learn more about this concept. 😉

**Reply by juancarlosjr97** [reply]
Thank you @viceice and @HonkingGoose ! I read the [workflow](https://docs.renovatebot.com/key-concepts/dashboard/#dependency-dashboard-approval-workflow) and I added them to my test project but it looks like because I am wrapping all as one single merge request it is not picking them up individually. https://github.com/juancarlosjr97/renovate-test-constraints/issues/3.

You can see the constraint here https://github.com/juancarlosjr97/renovate-test-constraints/blob/main/renovate.json#L13

**Reply by viceice** [reply]
add matchUpdateTypes=major and set group name=null

**Reply by juancarlosjr97** [reply]
That did the work. I put it as 

```
    "packageRules": [
        {
            "matchPackagePatterns": [
                "node-fetch"
            ],
            "dependencyDashboardApproval": true,
            "groupName": "devDependencies - node-fetch"
        }
    ]
```

All good, thanks!

**Reply by viceice** [reply]
that will also require approval for patch updates, I would suggest to limit it to major updates

**Reply by juancarlosjr97** [reply]
Thank you @viceice! I did the change adding only major updates but it does not pick up the node-fetch v2 as an automatic update. https://github.com/juancarlosjr97/renovate-test-constraints/issues/3

```
    "packageRules": [
        {
            "matchPackagePatterns": [
                "node-fetch"
            ],
            "dependencyDashboardApproval": true,
            "groupName": "devDependencies - node-fetch",
            "matchUpdateTypes": [
                "major"
            ]
        }
    ]
```

Thinking if I need a second rule for the v2 without `dependencyDashboardApproval` as `true`.

**Reply by viceice** [reply]
🤔 @rarkins any idea?

I would also set separate multiple major, but that needs a second rule because it can't be combined with match update types.

**Reply by juancarlosjr97** [reply]
Ok, I tried adding a second rule but not working as I would have expected... Any thoughts https://github.com/juancarlosjr97/renovate-test-constraints/blob/ea1507e6b71ba8369daf41053fd661269d137ab2/renovate.json?

**Reply by viceice** [reply]
🤔  what do you expect to happen there? you need to close and rename pr and delete branch to see the pending approval

**Reply by juancarlosjr97** [reply]
@viceice I did close them https://github.com/juancarlosjr97/renovate-test-constraints/pulls?q=is%3Apr+is%3Aclosed, but for the exercise, I closed the two most recent again and deleted the branches.

I am trying to get the update for the node-fetch v2. How can I add the rule for it as it is picking up node-fetch v3 but not the v2. I tried adding all the match types as you can see on my configuration file https://github.com/juancarlosjr97/renovate-test-constraints/blob/main/renovate.json#L25-L35

**Reply by juancarlosjr97** [reply]
So I went through the [logs](https://developer.mend.io/github/juancarlosjr97/renovate-test-constraints/-/job/018da821-e7ed-7afc-9697-738eb2daec38) and I could not find searching for v2, just when reaches to v3 it does not get more results when the v2 could be updated. So perhaps those two rules cannot work together?

**Reply by viceice** [reply]
reverse both rules and remove the approval and update types from the non-major rule.

you also need to rename closed PR so that renovate don't find them.

**Reply by viceice** [reply]
can you give me write access to the test repo, so I can see the renovate logs?

**Reply by juancarlosjr97** [reply]
Sure, access given @viceice :)

**Reply by juancarlosjr97** [reply]
Sorry @viceice , I did not follow the 

> reverse both rules and remove the approval and update types from the non-major rule.

Just in case. What I am trying to do is is pick up versions for node-fetch of the v2 but never the major, so never going to v3.
